### PR TITLE
Set decline code to PaymentRejectedException from CardException

### PR DESCRIPTION
**Pull request by artempWix**

### CODE REVIEW

**Comment by dleshem** [comment]
@artempWix this PR adds the decline code for `PaymentRejectedException`.

How about adding both code + decline_code for all failures, i.e. also for `PaymentErrorException`? This could help debug other failures as well.

I would set `gatewayInternalCode` to something like `(code == "card_declined" ? s"$code|$declineCode" : code)`

**Comment by artempWix** [comment]
@dleshem 
as i can see, `code` / `declineCode` exist only in `CardException` - so they are relevant only for `PaymentRejectedException` for now
`StripeException` has only `statusCode` with HTTP code - `libpay-stripe` is not receiving any other codes from Stripe SDK (except codes for `CardException`)
what should we put to `PaymentErrorException `?

about `code + decline_code`: as we can see in a [Stripe API Reference](https://stripe.com/docs/api#errors), `decline_code` is enough to identify the error. is it necessary to put `code` to the `gatewayInternalCode ` and conditionally (`code == "card_declined"`) parse it later?

**Comment by dleshem** [comment]
IIUC `decline_code` is only present in the response if `code` is `"card_declined"`. But there could be other codes, for example `"incorrect_zip"`.

**Comment by artempWix** [comment]
@dleshem 
are you sure? i have found no description of this case is a Stripe API Reference and i can't see this codes - we haven't received them before
but i suppose that `code == "card_declined"` is not only case when we have a `decline_code`

for example, one of possible values of `decline_code` and `code` both is `"expired_card"`
i can't imagine two different cases when we will receive next:
1. `code == "card_declined"`, `decline_code == "expired_card"`
2. `code == "expired_card"`, `decline_code == null`

what do you think?

**Comment by dleshem** [comment]
Stripe's actual implementation may differ from what's specified in their [API documentation](https://stripe.com/docs/api#errors-decline-code):

> **code** (_optional_): For card errors, a short string from amongst those listed on the right describing the kind of card error that occurred.
>
> **decline_code** (_optional_): For card errors resulting from a bank decline, a short string indicating the bank's reason for the decline if they provide one.

To me, this reads like `decline_code` is optional, and "by contract" limited to when `code` equals `"card_declined"`.

**Comment by artempWix** [comment]
@dleshem 
ok, actually i don't know what real values we are receiving from Stripe)
let's pass both codes in a format that you have proposed, i will add it to pull request

**Comment by dleshem** [comment]
Cool, looks good to me

File: libpay-stripe/src/main/scala/com/wix/pay/stripe/StripeGateway.scala [hunk 1]
```diff
@@ -130,13 +130,25 @@ class StripeGateway(merchantParser: StripeMerchantParser = new JsonStripeMerchan
...
   }
 
+  private def extractCardExceptionCode(e: CardException) = {
+    val code = Option(e.getCode)
+    val declineCode = Option(e.getDeclineCode)
+
+    (code, declineCode) match {
+      case (None, _) => None
+      case (_, None) => code
+      case (_, _) => Some(s"${code.get}|${declineCode.get}")
```
Comment (ykrasik) [inline libpay-stripe/src/main/scala/com/wix/pay/stripe/StripeGateway.scala]: let's use a destructuring declaration here:
`case (Some(code), Some(declineCode))`

**Pull request merged**